lib/index.js

```diff
@@ -57,9 +57,9 @@ internals.Client = class {
             options = Hoek.applyToDefaults(this._defaults, options, { shallow: internals.shallowOptions });
 
             Hoek.assert(options.payload === undefined || typeof options.payload === 'string' || typeof options.payload === 'object', 'options.payload must be a string, a Buffer, a Stream, or an Object');
-            Hoek.assert(options.agent === undefined || options.agent === null || typeof options.rejectUnauthorized !== 'boolean', 'options.agent cannot be set to an Agent at the same time as options.rejectUnauthorized is set');
-            Hoek.assert(options.beforeRedirect === undefined || options.beforeRedirect === null || typeof options.beforeRedirect === 'function', 'options.beforeRedirect must be a function');
-            Hoek.assert(options.redirected === undefined || options.redirected === null || typeof options.redirected === 'function', 'options.redirected must be a function');
+            Hoek.assert([null, undefined].includes(options.agent) || typeof options.rejectUnauthorized !== 'boolean', 'options.agent cannot be set to an Agent at the same time as options.rejectUnauthorized is set');
```

Comment:
do you think we should create a helper instead: isNullOrUndefined?

Reply:
Good call!  I took a pass at that 👍